src/sql/workbench/parts/objectExplorer/browser/serverTreeActionProvider.ts

```diff
@@ -180,7 +183,10 @@ export class ServerTreeActionProvider extends ContributableActionProvider {
 			this.addRestoreAction(context, actions);
 		}
 
-		actions.push(this._instantiationService.createInstance(RefreshAction, RefreshAction.ID, RefreshAction.LABEL, context.tree, treeNode));
+		// Contribute refresh action for scriptable objects via contribution
+		if (!this.isScriptableObject(context)) {
+			actions.push(this._instantiationService.createInstance(RefreshAction, RefreshAction.ID, RefreshAction.LABEL, context.tree, context.profile));
```

Comment:
Why do you need this and the contributed action?

Reply:
So that we don't have 2 refresh actions

Reply:
O you have a separate refresh action for scripting?

Reply:
Yes because it's it happens only for scriptable objects since we have contributed commands for those which ruin the order